android/engine/src/main/java/org/smartregister/fhircore/engine/task/WorkflowCarePlanGenerator.kt

```diff
@@ -167,15 +160,61 @@ constructor(
     )
   }
 
-  private fun createPlanDefinitionProcessor(): R4PlanDefinitionProcessor {
-    // val repository = getPrivateProperty("repository", fhirOperator) as ProxyRepository
-    // val evaluationSettings = getPrivateProperty("evaluationSettings", fhirOperator) as
-    // EvaluationSettings
+  private fun FhirOperator.generateCarePlan(
+    planDefinition: PlanDefinition,
+    subject: Patient,
+    data: Bundle,
+  ): IBaseResource {
+    // TODO: Open this method on the Android SDK
+    val planDefProcessor =
+      javaClass.getDeclaredField("planDefinitionProcessor").let {
+        it.isAccessible = true
+        return@let it.get(this) as PlanDefinitionProcessor
+      }
+
+    val libraryProcessor =
+      javaClass.getDeclaredField("libraryProcessor").let {
+        it.isAccessible = true
+        return@let it.get(this) as LibraryEngine
+      }
+
+    val params = Parameters()
+    params.addParameter(
+      part("%resource", data).apply {
+        extension.add(
+          Extension(
+            "http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-parameterDefinition",
+            ParameterDefinition().apply {
+              type = data.fhirType()
+              min = 0
+              max = data.entry.size.toString()
+            },
+          ),
+        )
+      },
+    )
+    params.addParameter(part("%rootResource", planDefinition))
+    params.addParameter(part("%subject", subject))
 
```

Comment:
Adds the objects for partial backwards compatibility with the Structure Map resolution. The PlanDefinitions would of course ONLY work in this project.